frontend/src/lib/components/Table/types.ts

```diff
@@ -0,0 +1,137 @@
+// };
+
+export type ColumnGroup<TData extends Record<string, any>> = {
+    _type: "group";
+    label: string;
+    columnId: string;
+    hoverText?: string;
+    sizeInPercent: number;
+    subColumns: TableColumn<TData>[];
+};
+
+export type TableColumn<TData extends Record<string, any>> = DataColumnAllKeys<TData> | ColumnGroup<TData>;
+//  | VirtualColumn<TData>
+
+export type TableColumns<TData extends Record<string, any>> = TableColumn<TData>[];
+
+export interface ColumnFilterImplementationProps<TFilterVal = any> {
+    value: TFilterVal | null | undefined;
+    onFilterChange: (newValue: TFilterVal | null | undefined) => void;
+}
+
+export type CustomColumnFilter<TData extends Record<string, any>, TDataKey extends keyof TData> = {
+    render?: (props: ColumnFilterImplementationProps<any>) => React.ReactNode;
+    predicate?: (
+        filterValue: any,
+        dataValue: TData[TDataKey],
+        dataDef: DataCellDef<TData, TDataKey>,
+        entry: TData,
+    ) => boolean;
+};
+
+export type PendingData = { _pending: true };
+export type LoadedData<TData extends Record<string, any>> = TData;
+export type TableData<TData extends Record<string, any>> = PendingData | LoadedData<TData>;
+
+export type PendingDataWithKey = { _key: string; _pending: true };
+export type LoadedDataWithKey<TData extends Record<string, any>> = { _key: string } & TData;
+export type TableDataWithKey<TData extends Record<string, any>> = PendingDataWithKey | LoadedDataWithKey<TData>;
+
+// --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---
+// - Table cell definitions -- --- --- --- --- --- --- --- --- --- --- --- ---
+export type HeaderCellDef = {
+    columnId: string;
+    colSpan: number;
+    rowSpan: number;
+    colGroupIndex: number;
+    sortable: boolean;
+    isGroup: boolean;
+    label: string;
+    hoverText?: string;
+};
+
+export type FilterCellDef<TData extends Record<string, any>> = {
+    columnId: string;
+    enabled: boolean;
+    colGroupIndex: number;
+    render?: CustomColumnFilter<TData, any>["render"];
+};
+
+export type DataCellDef<TData extends Record<string, any>, TK extends keyof TData> = {
+    columnId: keyof TData;
+    colGroupIndex: number;
+    format?: (value: TData[TK], entry: TData) => string;
+    style?: (value: TData[TK], entry: TData) => React.CSSProperties;
+    render?: (value: TData[TK], entry: TData) => React.ReactNode;
+    filter?: CustomColumnFilter<TData, TK>["predicate"];
+};
+
+export type TableCellDefinitions<TData extends Record<string, any>> = {
+    headerCells: HeaderCellDef[][];
+    filterCells: FilterCellDef<TData>[];
+    dataCells: { [TK in keyof TData]: DataCellDef<TData, TK> }[keyof TData][];
+};
+
+// --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---
+// - Table collation definitions - --- --- --- --- --- --- --- --- --- --- ---
+export enum SortDirection {
+    ASC = "asc",
+    DESC = "desc",
```

Comment:
Consider adding doc that "asc" and "desc" enum string values is chosen as this is used/passed to lodash `sortBy`-function in code?